pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/flags.go

```diff
@@ -0,0 +1,271 @@
+	args := map[string][]string{}
+	for key, slice := range kubeAPIServerConfig.KubernetesMasterConfig.APIServerArguments {
+		for _, val := range slice {
+			args[key] = append(args[key], val)
+		}
+	}
+
+	return args
+}
+
+func setIfUnset(cmdLineArgs map[string][]string, key string, value ...string) {
+	if _, ok := cmdLineArgs[key]; !ok {
+		cmdLineArgs[key] = value
+	}
+}
+
+func admissionFlags(kubeAPIServerConfig *configapi.MasterConfig) (map[string][]string, error) {
+	args := map[string][]string{}
+
+	forceOn := []string{}
+	forceOff := []string{}
+	pluginConfig := map[string]configapi.AdmissionPluginConfig{}
+	for pluginName, config := range kubeAPIServerConfig.AdmissionConfig.DeepCopy().PluginConfig {
+		if len(config.Location) > 0 {
+			content, err := ioutil.ReadFile(config.Location)
+			if err != nil {
+				return nil, err
+			}
+			// if the config isn't a DefaultAdmissionConfig, then assume we're enabled (we were called after all)
+			// if the config *is* a DefaultAdmissionConfig and it explicitly said
+			obj, err := configapilatest.ReadYAML(bytes.NewBuffer(content))
+			// if we can't read it, let the plugin deal with it
+			// if nothing was there, let the plugin deal with it
+			if err != nil || obj == nil {
+				forceOn = append(forceOn, pluginName)
+				config.Location = ""
+				config.Configuration = &runtime.Unknown{Raw: content}
+				pluginConfig[pluginName] = *config
+				continue
+			}
+
+			if defaultConfig, ok := obj.(*configapi.DefaultAdmissionConfig); !ok {
+				forceOn = append(forceOn, pluginName)
+				config.Location = ""
+				config.Configuration = &runtime.Unknown{Raw: content}
+				pluginConfig[pluginName] = *config
+				continue
+
+			} else if defaultConfig.Disable {
+				forceOff = append(forceOff, pluginName)
+				continue
+			} else {
+				forceOn = append(forceOn, pluginName)
+				continue
+			}
+
+			continue
+		}
+		// if it wasn't a DefaultAdmissionConfig object, let the plugin deal with it
+		if defaultConfig, ok := config.Configuration.(*configapi.DefaultAdmissionConfig); !ok {
+			forceOn = append(forceOn, pluginName)
+			pluginConfig[pluginName] = *config
+			continue
+
+		} else if defaultConfig.Disable {
+			forceOff = append(forceOff, pluginName)
+			continue
+		} else {
+			forceOn = append(forceOn, pluginName)
+			continue
+		}
+
+	}
+	upstreamAdmissionConfig, err := originadmission.ConvertOpenshiftAdmissionConfigToKubeAdmissionConfig(pluginConfig)
+	if err != nil {
+		return nil, err
+	}
+	configBytes, err := configapilatest.WriteYAML(upstreamAdmissionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	tempFile, err := ioutil.TempFile("", "kubeapiserver-admission-config.yaml")
+	if err != nil {
+		return nil, err
+	}
+	if _, err := tempFile.Write(configBytes); err != nil {
+		return nil, err
+	}
+	tempFile.Close()
+
+	setIfUnset(args, "admission-control-config-file", tempFile.Name())
+	setIfUnset(args, "disable-admission-plugins", forceOff...)
```

Comment:
these are additive to the default values. Maybe we should append?

Reply:
> these are additive to the default values. Maybe we should append?

We've always stomped admission with our choices.  I only want one way to set via config.